Refactor and Expand

netzhuffle opened the pull request:
Refactored, unified code style, fixed #17 and added 66 new insults from http://geekdad.com/2013/03/66-geeky-insults-you-can-use-almost-anywhere/.

netzhuffle committed:
Fixes #17

netzhuffle committed:
Refactor and unify code style

Added JSDoc for all functions
Configured JSCS to random preferred code style
Small change: The bots name is now case insensitive

netzhuffle committed:
66 new insults, yay!

Thanks to Jason CranfordTeague at http://geekdad.com/2013/03/66-geeky-insults-you-can-use-almost-anywhere/.

netzhuffle committed:
Correct Escape from Monkey Island quotes

netzhuffle committed:
Two new stickers from Swiss quotes set

dbrgn commented:
@rnestler maybe? :)

netzhuffle commented on one_liners.js:
```diff
@@ -0,0 +1,172 @@
+ */
+exports.getRandomInsult = function(userName) {
+  var insult = insults[Math.floor(Math.random() * insults.length)];
+  return insult.replace('%u%', userName);
```
Bug: `'%u%'` has to be `/%u%/g` to replace all concurrencies of %u% in the string.

dns2utf8 commented on bot.js:
```diff
@@ -1,140 +1,72 @@
- * @returns {string} The insult
  */
 function replyInsult(message, user) {
   if (Math.random() * 20 > 1) {
```
Is this 20% chance? Maybe write it like this:

```
if (Math.random() < 0.2)
```

rnestler replied:
And now it's all clear why it insults so often ^^ Random \* 20 leads to evenly distributed values [0..20] which are > 1 in 19 of 20 cases which is ~95%

netzhuffle replied:
This isn’t the insult chance, it is the chance of text insult versus Telegram sticker insult. When replyInsult is called, the bot already decided it’s going to insult.

dns2utf8 commented on one_liners.js:
```diff
@@ -0,0 +1,172 @@
+  'BQADBAADOQADmu78Ap0s66P75zWsAg', // Toni Brunner 😮
+  'BQADBAADRwADmu78AklGZ2FcPSkKAg', // Ueli Maurer 😀
+  'BQADBAADrQADEhKdAAG75Oi8qTO6fwI', // 👐 Chabis
+  'BQADBAADwQADEhKdAAFk3LSH1pscvgI', // 😈 Verzells am Chääs
```
Maybe add:

``` js
'BQADBAADNBAAAiGJygABwh0LudHRUtMC' // Verzell kein Salat
```

dns2utf8 commented on replies.js:
```diff
@@ -0,0 +1,266 @@
+module.exports.search = function(query) {
+  var match = fuzzy.get(query);
+  if (!match) {
+    return null;
```
`null` vs. `throw`

netzhuffle replied:
„throw“ would be for exceptions – but not finding a reply when it’s not a monkey island insult is expected behavior. I think “null” is a meaningful return value for saying “the search found no no reply”. Do you see that differently?

dns2utf8 replied:
Ah, I interpreted it as `findOne`. 

I personaly would return `undefined` not `null` because `null` leads to problems: http://saladwithsteve.com/2008/02/javascript-undefined-vs-null.html

dns2utf8 commented:
Can this branch be rebased to mastter after #19 ?

netzhuffle committed:
Refactor

Based on code review by @dns2utf8
and Airbnb JavaScript Style Guide (https://github.com/airbnb/javascript)

netzhuffle commented:
@dns2utf8 Mind checking again & merging if everything looks good?

dns2utf8 commented on bot.js:
```diff
@@ -1,147 +1,79 @@
 }
 
-bot.on('message', function(message) {
+bot.on('message', (message) => {
```
Fancy :+1:

dns2utf8 commented on replies.js:
```diff
@@ -0,0 +1,271 @@
+
+// jscs:enable requireTrailingComma
+
+const fuzzy = FuzzySet([...insults.keys()]);
```
What does this line do? Maybe add a comment to the array-access.

I read in the manual:
Read-only elements that specify the parenthesized substring matches, if included in the regular expression. The number of possible parenthesized substrings is unlimited.

netzhuffle replied:
This line is not an array access, but a new array construction. The spread operator “`...`” is to convert an iterable expression to multiple parameters. The function `insults.keys()` returns an iterator but `FuzzySet` needs an array parameter. This is the default way to create an array from an iterator in EcmaScript 2015: The spread operator converts each iterator value to a parameter for the array constructor. The line is equivalent to the following:

`const keyIterator = insult.keys();
const keyArray = [...keyIterator];
const fuzzy = FuzzySet(keyArray);`

`[...keyIterator]` is the same thing as writing `[keyIterator.next(), keyIterator.next(),  keyIterator.next(),` (and so on, as long as there is a next value) `]` with multiple parameters by hand, where `[param1, param2, paramn]` is the normal array constructor.

As the spread operator is plain standard EcmaScript 2015 (albeit ES2015 is fairly new) and this is the default way to create an array from an iterator, I don’t see the need to add a comment here. Agree, @dns2utf8?

dbrgn replied:
+1 :)

dns2utf8 commented:
Maybe add a comment to the array-access, apart from that :rainbow:
